### PR TITLE
fix: sdk 50 tests #NTRN-241

### PR DESCRIPTION
**Pull request by swelf19**
All unit tests work.

removed ccv reward migration and test. since the module moved params into own store and migration is not applicable anymore

To successfully pass the tests it's required to fix provider app initilization (https://github.com/neutron-org/interchain-security/commit/bf0abfba3230857ddfe69a9cd4028182d1254e78)

### CODE REVIEW

File: network/start.sh [hunk 1]
```diff
@@ -19,7 +19,6 @@ if [ "$RUN_BACKGROUND" == 1 ]; then
     --home "$CHAIN_DIR"                   \
     --pruning=nothing                     \
     --grpc.address="0.0.0.0:$GRPCPORT"    \
-    --grpc-web.address="0.0.0.0:$GRPCWEB" \
```
Comment (mmikev) [inline network/start.sh:22]: Why is that?

File: testutil/apptesting/test_suite.go [hunk 2]
```diff
@@ -37,12 +38,10 @@ type KeeperTestHelper struct {
 // Setup sets up basic environment for suite (App, Ctx, and test accounts)
 func (s *KeeperTestHelper) Setup() {
 	s.App = testutil.Setup(s.T()).(*app.App)
-	ctx := s.App.GetBaseApp().NewContext(
-		false,
-	)
+	ctx := s.App.GetBaseApp().NewUncachedContext(false, tmtypes.Header{Height: 1, ChainID: "neutron-1", Time: time.Now().UTC()})
```
Comment (mmikev) [inline testutil/apptesting/test_suite.go]: Could you add a small comment here that describes why do we need such complex context?

File: x/dex/keeper/core_helper_test.go [hunk 3]
```diff
@@ -29,7 +30,7 @@ func TestCoreHelpersTestSuite(t *testing.T) {
 
 func (s *CoreHelpersTestSuite) SetupTest() {
 	app := testutil.Setup(s.T())
-	ctx := app.(*neutronapp.App).BaseApp.NewContext(false)
+	ctx := app.(*neutronapp.App).BaseApp.NewUncachedContext(false, cmtproto.Header{})
```
Comment (mmikev) [inline x/dex/keeper/core_helper_test.go:37]: Same here about context creation

File: x/dex/keeper/grpc_query_user_deposits_test.go [hunk 4]
```diff
@@ -32,7 +33,7 @@ func simulateDeposit(ctx sdk.Context, app *neutronapp.App, addr sdk.AccAddress,
 func TestUserDepositsAllQueryPaginated(t *testing.T) {
 	app := testutil.Setup(t)
 	keeper := app.(*neutronapp.App).DexKeeper
-	ctx := app.(*neutronapp.App).BaseApp.NewContext(false)
+	ctx := app.(*neutronapp.App).BaseApp.NewUncachedContext(false, tmproto.Header{})
```
Comment (mmikev) [inline x/dex/keeper/grpc_query_user_deposits_test.go:41]: And here, pls

File: x/dex/keeper/integration_placelimitorder_test.go [hunk 5]
```diff
@@ -653,8 +653,9 @@ func (s *DexTestSuite) TestPlaceLimitOrderGoodTilExpiresNotPurged() {
 	s.assertAliceBalances(0, 0)
 
 	// When two days go by
-	// for simplicity sake we never run endBlock, it reality it would be run, but gas limit would be hit
```
Comment (mmikev) [inline x/dex/keeper/integration_placelimitorder_test.go:656]: i guess you've removed this comment by mistake

File: x/dex/keeper/integration_cancellimitorder_test.go [hunk 6]
```diff
@@ -249,7 +249,6 @@ func (s *DexTestSuite) TestCancelGoodTilAfterExpirationFails() {
 
 	// WHEN expiration date has passed
 	s.nextBlockWithTime(time.Now().AddDate(0, 0, 2))
-	// TODO: s.App.EndBlock(abci.RequestEndBlock{Height: 0})
```
Comment (mmikev) [inline x/dex/keeper/integration_cancellimitorder_test.go:252]: I remember you talked about inability to call endblock in Cosmos SDK 0.50. How did you manage to resolve that?

File: x/dex/keeper/msg_server_test.go [hunk 7]
```diff
@@ -1573,6 +1575,15 @@ func (s *DexTestSuite) nextBlockWithTime(blockTime time.Time) {
 	newCtx := s.Ctx.WithBlockTime(blockTime)
 	s.Ctx = newCtx
 	s.GoCtx = newCtx
+	_, err := s.App.FinalizeBlock(&abci.RequestFinalizeBlock{
+		Height: s.App.LastBlockHeight() + 1,
+		Time:   blockTime,
+		//NextValidatorsHash: s.App.GetValidatorSet(s.Ctx).,
```
Comment (mmikev) [inline x/dex/keeper/msg_server_test.go]: do we need this?

File: x/interchainqueries/keeper/process_block_results_test.go [hunk 8]
```diff
@@ -115,16 +126,14 @@ func NextBlock(chain *ibctesting.TestChain) {
...
-	// TODO: chain.App.BeginBlock(abci.RequestBeginBlock{Header: chain.CurrentHeader})
 }
 
 // CommitBlock commits a block on the provided indexes and then increments the global time.
 //
 // CONTRACT: the passed in list of indexes must not contain duplicates
 func CommitBlock(coord *ibctesting.Coordinator, chains ...*ibctesting.TestChain) {
 	for _, chain := range chains {
-		chain.App.Commit()
+		//chain.NextBlock()
```
Comment (mmikev) [inline x/interchainqueries/keeper/process_block_results_test.go]: i guess we can safely remove it

**Review by pr0n00gler — CHANGES_REQUESTED** [review]
Left some comments/questions.
Plus please resolve some linter issues

File: network/start.sh (hunk 1 above)
Comment (swelf19) [reply network/start.sh:22]: there is no such option anymore

File: x/dex/keeper/integration_cancellimitorder_test.go (hunk 6 above)
Comment (swelf19) [reply x/dex/keeper/integration_cancellimitorder_test.go:252]: it was about `TestPlaceLimitOrderGoodTilExpiresNotPurged` test. I needed to move chain forward without calling endblocker. But it turned out, it's enough just to change time in `ctx` to simulate that chain is alive.

**Review by pr0n00gler — APPROVED** [review]

**Pull request merged**